src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1569 @@
+                //         _ = a is ([>0]) and ([<0]);   // 5
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is ([>0]) and ([<0])").WithArguments("int[]").WithLocation(15, 13)
+                );
+        }
+
+        [Fact]
+        public void ListPattern_Symbols()
+        {
+            var source =
+@"class X
+{
+    public void Test1()
+    {
+        _ = new string[0] is [var length1];
+        _ = new string[0] is {var element1};
+        _ = new string[0] is {..var slice1};
+
+        _ = new int[0] is [var length2];
+        _ = new int[0] is {var element2};
+        _ = new int[0] is {..var slice2};
+
+        _ = new string[0] is [int length3];
+        _ = new string[0] is {string element3};
+        _ = new string[0] is {..string[] slice3};
+
+        _ = new int[0] is [int length4];
+        _ = new int[0] is {int element4};
+        _ = new int[0] is {..int[] slice4};
+    }
+}";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithListPatterns);
+            compilation.VerifyDiagnostics();
+            var tree = compilation.SyntaxTrees[0];
+            var nodes = tree.GetRoot().DescendantNodes().OfType<SingleVariableDesignationSyntax>();
+            Assert.Collection(nodes,
+                d => verify(d, "var length1", "int", "int"),
+                d => verify(d, "var element1", "string?", "string"),
+                d => verify(d, "var slice1", "string[]?", "string[]"),
+
+                d => verify(d, "var length2", "int", "int"),
+                d => verify(d, "var element2", "int", "int"),
+                d => verify(d, "var slice2", "int[]?", "int[]"),
+
+                d => verify(d, "int length3", "int", "int"),
+                d => verify(d, "string element3", "string", "string"),
+                d => verify(d, "string[] slice3", "string[]", "string[]"),
+
+                d => verify(d, "int length4", "int", "int"),
+                d => verify(d, "int element4", "int", "int"),
+                d => verify(d, "int[] slice4", "int[]", "int[]")
+            );
+
+            void verify(SyntaxNode designation, string syntax, string declaredType, string type)
+            {
+                Assert.Equal(syntax, designation.Parent.ToString());
+                var model = compilation.GetSemanticModel(tree);
+                var symbol = model.GetDeclaredSymbol(designation);
+                Assert.Equal(SymbolKind.Local, symbol.Kind);
+                Assert.Equal(declaredType, ((ILocalSymbol)symbol).Type.ToDisplayString());
+                var typeInfo = model.GetTypeInfo(designation);
+                Assert.Null(typeInfo.Type);
+                Assert.Null(typeInfo.ConvertedType);
+                typeInfo = model.GetTypeInfo(designation.Parent);
+                Assert.Equal(type, typeInfo.Type.ToDisplayString());
+                Assert.Equal(type, typeInfo.ConvertedType.ToDisplayString());
+            }
+        }
```

Comment:
Another test case idea for `GetTypeInfo`: `is { ..{} }`
On the inner braces it should give the input type for the nested pattern (coming from the slice pattern).